packages/calypso-e2e/src/lib/pages/media-page.ts

```diff
@@ -140,4 +147,30 @@ export class MediaPage extends BaseContainer {
 		await this.page.click( selectors.imageEditorCancelButton );
 		await this.page.waitForSelector( selectors.mediaModal );
 	}
+
+	/**
+	 * Uploads the file to the Media gallery.
+	 *
+	 * @param {string} fullPath Full path to the file on disk.
+	 * @returns {Promise<void>} No return value.
+	 */
+	async upload( fullPath: string ): Promise< ElementHandle > {
+		await this.page.setInputFiles( selectors.addNewButton, fullPath );
+
+		const filename = path.basename( fullPath );
+		const itemSelector = `figure[title="${ filename }"]`;
+		const result = await Promise.race( [
+			this.page.waitForSelector( itemSelector ), // Upload successful
+			this.page.waitForSelector( selectors.uploadRejectionNotice ), // Upload failed
+		] );
+
+		if ( ( await result.getAttribute( 'title' ) ) === filename ) {
+			return result;
+		}
+
+		const noticeText = await this.page
+			.waitForSelector( '.notice__text' )
+			.then( ( element ) => element?.innerText() );
+		throw new Error( noticeText );
```

Comment:
In this case the result will be the notice, so we can go directly for its content:

```suggestion
		throw new Error( result.innerText() );
```